slave.mk

```diff
@@ -664,6 +665,103 @@ $(addprefix $(TARGET_PATH)/, $(SONIC_INSTALLERS)) : $(TARGET_PATH)/% : \
 
 SONIC_TARGET_LIST += $(addprefix $(TARGET_PATH)/, $(SONIC_INSTALLERS))
 
+# targets for building debug installers with base image
+$(addprefix $(TARGET_PATH)/, $(SONIC_DBG_INSTALLERS)) : $(TARGET_PATH)/$(DBG_IMAGE_MARK)-% : \
```

Comment:
Hi Marian,

Yes indeed this is one of the final pieces of the work, with debug dockers. 

Main Goal:
To transparently build debug docker images for each docker image. 

Gain/purpose of the goal:
Today, if you need to a debug session, on a dead core file or live debugging, you need to find DUT running that version, get into docker container, download & install all debug symbols and download/install debug tools like gdb.

If you are looking at core files from two different versions, you need two  DUTs (virtual/real)

With this goal:
To analyze a dead core file, you *don't* need any DUT. From the build server, get debug version of your docker, run the docker in any docker-supported environment (e.g. a VM running linux), load & run the docker with bash as entrypoint and core file mapped.  
This saves us a lot of time and reduce our requirements.

To do live debugging, instead of upgrading your docker  with debug symbols & tools, get the debug version of the same docker from the build server, load that image and you are set to go.

BTW, this main goal is already accomplished for all *stretch* based dockers, through following PRs that are committed.

https://github.com/Azure/sonic-buildimage/pull/2753
https://github.com/Azure/sonic-buildimage/pull/2789
https://github.com/Azure/sonic-buildimage/pull/2812

Add-on goal:
Now the debug docker images are available for all stretch based dockers. With goal of moving all dockers to stretch based, soon all docker images will have a debug version available.

So wonder, why not build a debug version of final sonic image (.bin/.swi/...) using debug docker images.

With this, you could have a real/virtual DUT with all docker images enhanced with debug symbols & tools.

This PR is the first step, to build debug version of sonic-broadcom.bin.

Reply:
This is a version for debug image.
It uses all DBG Dockers.

Reply:
I agree with marian, too many duplicated with the non debug version. Need to simplify.

Reply:
Agree! I have been thinking too in the same line. Let me try to merge and as part make the "-dbg" as suffix too.

Reply:
Done